spacer/extract_features_utils.py

```diff
@@ -60,3 +60,20 @@ def crop_simple(im: np.ndarray,
     left = int(center[1] - crop_size / 2)
 
     return im[upper: upper + crop_size, left: left + crop_size, :]
+
+
+def crop_patches_pil(im: Image,
+                     rowcols: List[Tuple[int, int]],
+                     crop_size: int):
+    pad = crop_size
+    image = Image.fromarray(
+        np.pad(im.convert('RGB'),
+               ((pad, pad), (pad, pad), (0, 0)), mode='reflect'),
+        'RGB'
+    )
+
+    return [image.crop((pad + col - crop_size / 2,
```

Comment:
Image.crop returns Image objects, so you'd need to convert to arrays here to match with the previous method. https://pillow.readthedocs.io/en/3.1.x/reference/Image.html#PIL.Image.Image.crop

Reply:
if we convert to arrays here, wouldn't it be the same as the original `crop_patches`?

In Caffe, we have `Transformer` that can process `Image` object https://github.com/beijbom/pyspacer/blob/b106de785e21e14cfbac03878ff85f1b48b80847/spacer/caffe_utils.py#L26-L34
In Pytorch, we have `transformation` that can process `Image` object as well
https://github.com/beijbom/pyspacer/blob/b106de785e21e14cfbac03878ff85f1b48b80847/spacer/torch_utils.py#L15-L24